itests/deals_concurrent_test.go

```diff
@@ -47,3 +57,71 @@ func TestDealCyclesConcurrent(t *testing.T) {
 		t.Run(ns+"-stdretrieval-NoCAR", func(t *testing.T) { runTest(t, n, false, false) })
 	}
 }
+
+func TestSimultenousTransferLimit(t *testing.T) {
+	if testing.Short() {
+		t.Skip("skipping test in short mode")
+	}
+
+	kit.QuietMiningLogs()
+
+	blockTime := 10 * time.Millisecond
+
+	// For these tests where the block time is artificially short, just use
+	// a deal start epoch that is guaranteed to be far enough in the future
+	// so that the deal starts sealing in time
+	startEpoch := abi.ChainEpoch(2 << 12)
+
+	runTest := func(t *testing.T) {
+		client, miner, ens := kit.EnsembleMinimal(t, kit.MockProofs(), kit.ConstructorOpts(
+			node.ApplyIf(node.IsType(repo.StorageMiner), node.Override(new(dtypes.StagingGraphsync), modules.StagingGraphsync(2))),
+		))
+		ens.InterconnectAll().BeginMining(blockTime)
+		dh := kit.NewDealHarness(t, client, miner)
+
+		ctx, cancel := context.WithCancel(context.Background())
+
+		du, err := miner.MarketDataTransferUpdates(ctx)
+		require.NoError(t, err)
+
+		var maxOngoing int
+		var wg sync.WaitGroup
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+
+			ongoing := map[datatransfer.TransferID]struct{}{}
+
+			for {
+				select {
+				case u := <-du:
+					t.Logf("%d - %s", u.TransferID, datatransfer.Statuses[u.Status])
+					if u.Status == datatransfer.Ongoing {
+						ongoing[u.TransferID] = struct{}{}
+					} else {
+						delete(ongoing, u.TransferID)
+					}
+
+					if len(ongoing) > maxOngoing {
+						maxOngoing = len(ongoing)
+					}
+				case <-ctx.Done():
+					return
+				}
+			}
+		}()
+
+		dh.RunConcurrentDeals(kit.RunConcurrentDealsOpts{
+			N:             1, // TODO: set to 20 after https://github.com/ipfs/go-graphsync/issues/175 is fixed
```

Comment:
With this set to 10, when the test would not hang because of this issue, the assert below would fail if the config wasn't set, and it does pass when it's set